src/main/java/com/worldwarofants/game/util/RandGen.java

```diff
@@ -0,0 +1,77 @@
+package com.worldwarofants.game.util;
+
+import java.util.Random;
+
+/*
+    Utility class that provides an abstraction of Java's Random class.
+    Its goal is to provide developers with a simple way to work with
+    random number generation, as well as allow for precise tweaking
+    of exact methods used.
+
+    Please make sure to generate your random numbers within the project
+    exclusively using this package.
+*/
+
+public class RandGen extends java.util.Random {
+    private static final long serialVersionUID = -5378246343912387722L;
+
+    Random generator;
+    long seed;
+    String seedString; // If a custom text input has been provided by user and hashed,
+
+    // Overloaded seedless constructor
+    public RandGen() {
+        this.generator = new Random(); // Create a generator
+        this.seed = generator.nextLong(); // Ask it to generate a seed for itself
+        this.seedString = Long.toString(this.seed); // Save it to seedString as well
+        this.generator.setSeed(seed); // And set its seed to what it gave
+    }
+
+    // Overloaded seeded (with a string) constructor
+    public RandGen(String seedString) {
+        if (seedString.matches("-?\\d+(\\.\\d+)?")) { // If the provided string is a valid long, use it as the seed
+            this.seed = Long.parseLong(seedString);
+            this.seedString = Long.toString(this.seed);
+        } else { // But if not, hash and record it
+            this.seed = seedString.hashCode();
+            this.seedString = seedString;
+        }
+        this.generator = new Random(seed);
+    }
+
+    // Return seed used
+    public long getSeed() {
```

Comment:
Why do you think the user of this class needs to know what the seed and the seedString is? What would be the point? Don't add getters without a need for them. Only when they are very necessary. And even then, think of how you can avoid doing it, before actually doing it.

Reply:
For bookkeeping purposes.
Suppose you stumbled upon a seed that produces an interesting result, or a bug.
Being able to access the seed responsible for purposes of sharing is the use case for this.